dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/header/HeaderExchangeHandler.java

```diff
@@ -209,10 +209,21 @@ public void received(Channel channel, Object message) throws RemotingException {
                     Exception e = new Exception("Dubbo client can not supported string message: " + message + " in channel: " + channel + ", url: " + channel.getUrl());
                     logger.error(e.getMessage(), e);
                 } else {
-                    String echo = handler.telnet(channel, (String) message);
-                    if (echo != null && echo.length() > 0) {
-                        channel.send(echo);
+                    String enableTelnetParam = channel.getUrl().getParameter(Constants.TELNET_ENABLE);

```

Comment:
boolean enableTelnet = channel.getUrl().getParameter(Constants.TELNET_ENABLE, true);

Reply:
got it